src/poprox_recommender/components/joiners/wrrf.py

```diff
@@ -0,0 +1,77 @@
+from collections import defaultdict
+from itertools import zip_longest
+from typing import List, Optional
+
+from lenskit.pipeline import Component
+from pydantic import BaseModel
+
+from poprox_concepts.domain import RecommendationList
+
+
+class WRRFConfig(BaseModel):
+    num_slots: int
+    k: int = 60
+    # Default weights if not specified
+    default_weights: dict = {"recs1": 1.0, "recs2": 1.0}
+
+
+class WeightedReciprocalRankFusion(Component):
+    config: WRRFConfig
+
+    def __call__(
+        self,
+        *,
+        recs1: RecommendationList,
+        recs2: RecommendationList,
+        additional_recs: Optional[List[RecommendationList]] = None,
+        weights: Optional[dict] = None,
```

Comment:
Do we envision a component computing weights such that we want this to be a runtime input, rather than just supporting it as a configuration option?